### PR TITLE
User: Make user email read-only

### PR DESCRIPTION
**Pull request by k9845**

### CODE REVIEW

**Review by tnagorra — APPROVED** [review]

**Review by annshress — APPROVED** [review]

File: apps/user/serializers.py [hunk 1]
```diff
@@ -71,6 +75,7 @@ class Meta:
                   'login_attempts', 'recaptcha_response',
                   'email', 'organization', 'display_picture',
                   'language', 'email_opt_outs')
+        write_only_on_create_fields = ('email', )
```
Comment (thenav56) [inline apps/user/serializers.py]: We also need `username`
```suggestion
        write_only_on_create_fields = ('email', 'username',)
```

File: apps/user/serializers.py [hunk 2]
```diff
@@ -2,7 +2,11 @@
 from drf_dynamic_fields import DynamicFieldsMixin
 from rest_framework import serializers
 
-from deep.serializers import RemoveNullFieldsMixin, URLCachedFileField
+from deep.serializers import (
+    RemoveNullFieldsMixin,
+    URLCachedFileField,
+    WriteOnlyOnCreateSerializerMixin
```
Comment (thenav56) [inline apps/user/serializers.py]:
```suggestion
    WriteOnlyOnCreateSerializerMixin,
```

File: apps/user/tests/test_apis.py [hunk 3]
```diff
@@ -34,7 +34,24 @@ def test_create_user(self):
...
+        self.assertEqual(r_data['username'], data['username'])
+
+        data1 = {
+            'email': 'hari.krishna@gmail.com'
+        }
+        user= User.objects.get(id=r_data['id'])
+        self.authenticate(user=user)
+        response = self.client.patch(f"{url}{r_data['id']}/", data1)
+        self.assert_200(response)  # authenticate user with same user id
+        self.assertEqual(response.data['email'], data['email'])  #should return previous email
```
Comment (thenav56) [inline apps/user/tests/test_apis.py]:
```suggestion
        self.assertEqual(response.data['email'], data['email'])  # Should return previous email
```

File: apps/user/tests/test_apis.py [hunk 4]
```diff
@@ -34,7 +34,24 @@ def test_create_user(self):
...
+
+        data1 = {
+            'email': 'hari.krishna@gmail.com'
+        }
+        self.authenticate()
+        response = self.client.patch(f"{url}{r_data['id']}/", data1)
+        self.assert_403(response)  #authenticate user with different user id
```
Comment (thenav56) [inline apps/user/tests/test_apis.py]:
```suggestion
        self.assert_403(response)  # aAuthenticate user with different user id
```

File: apps/user/tests/test_apis.py (hunk 3 above)
Comment (thenav56) [inline apps/user/tests/test_apis.py]: Also, check for `username`.

**Review by annshress — APPROVED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**